src/main/java/com/bringframework/registry/BeanDefinitionRegistryImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.bringframework.registry;
+
+import com.bringframework.exceptions.BeanDefinitionDuplicateNameException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Base implementation of {@link BeanDefinitionRegistry} used to store {@link BeanDefinition} and
+ * retrieve them. This implementation is thread-safe.
+ */
+@Slf4j
+public class BeanDefinitionRegistryImpl implements BeanDefinitionRegistry {
```

Comment:
I saw another naming for implementations like  `DefaultBeanDefinitionReader`, so let's align naming during daily

Reply:
Renamed to DefaultBeanDefinitionRegistry